staging/src/k8s.io/apiserver/pkg/storage/names/generate_test.go

```diff
@@ -19,11 +19,50 @@ package names
 import (
 	"strings"
 	"testing"
+
+	utilrand "k8s.io/apimachinery/pkg/util/rand"
 )
 
 func TestSimpleNameGenerator(t *testing.T) {
-	name := SimpleNameGenerator.GenerateName("foo")
-	if !strings.HasPrefix(name, "foo") || name == "foo" {
-		t.Errorf("unexpected name: %s", name)
+	mkBase := func(n int) string {
+		return utilrand.String(n-1) + "-"
```

Comment:
I suggest using strings.Repeat of a character that's not in the alphabet, so that there can't be flakes by it happening to pick the same character in the off-by-one test. (only needed if you take my other suggestion below)

Reply:
I did random so prefix-testing was more reliable.  Take a look at last commit.